google/cloud/storage/testbench/testbench.py

```diff
@@ -1345,6 +1350,9 @@ def buckets_insert():
     if bucket_name is None:
         raise error_response.ErrorResponse(
             'Missing bucket name in `Buckets: insert`', status_code=412)
+    if len(bucket_name) > 63 or not re.match('^[a-z][a-z-]*[a-z]', bucket_name):
```

Comment:
I think this regex is incorrect, given the guidelines at https://cloud.google.com/storage/docs/naming.  I actually had a snippet laying around that covers the checks mentioned in our docs, if you'd rather use that (probably worth factoring out into its own method):

```
valid = True
if '.' in bucket_name:
    valid &= (
        len(bucket_name) <= 222 and
        all([len(part) <= 63 for part in bucket_name.split('.')]))
else:
    valid &= len(bucket_name) <= 63
valid &= re.match('^[a-z0-9][a-z0-9._-]+[a-z0-9]$', bucket_name)
valid &= not bucket_name.startswith('goog')
valid &= re.search('g[0o][0o]g[1l][e3]', bucket_name)
valid &= re.match('^[0-9]{1,3}[.][0-9]{1,3}[.][0-9]{1,3}[.][0-9]{1,3}$', bucket_name)
```

The snippet above allows underscores -- while we discourage their use, they're still technically allowed in (GCS) bucket names.

Reply:
Done.